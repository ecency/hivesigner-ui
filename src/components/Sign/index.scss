.SignContainer {
  margin-top: 10vh;
}
.SignHeader {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  @media screen and (max-width: 630px) {
    padding: 0 10px;
  }
  .homelogo {
    width: auto;
    height: auto;
  }
}
.Title {
  font-size: 4rem;
  color: red;
}
.SignMenubar {
  font-size: 2rem;
  color: black;
  position: absolute;
  right: 30px;
  top: 50px;
  display: block;
  @media screen and (min-width: 630px) {
    display: none;
  }
}

.TransSearch {
  display: flex;
  justify-content: center;
  height: 70px;
  padding: 10px 0;
  width: 100%;
  @media screen and (max-width: 630px) {
    width: 100%;
    margin-left: 5px;
  }
  .icon {
    margin-top: 15px;
    z-index: 999;
    margin-left: -35px;
    font-size: 25px;
    color: #828282;
  }
  input {
    height: 55px;
    border: 1px solid #c4c4c4;
    border-radius: 5px;
    width: 600px;
    padding: 0 40px 0 15px;
    transition: all 0.3s;
    @media screen and (max-width: 630px) {
      width: 100%;
    }
  }
  input:focus,
  input:active {
    outline: none !important;
    border: 1px solid red;
  }
}
.Signfix {
  display: flex;
  height: 100vh;
  align-items: center;
  justify-content: center;
  margin: auto;
  height: 100px;
  position: fixed;

  bottom: 0 !important;
  background-color: white;
  width: 100%;
  * {
    top: 10px !important;
  }
}
.ResultContent {
  margin-bottom: 100px;
}
