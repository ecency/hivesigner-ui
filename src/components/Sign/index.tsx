import React, { useState, useEffect } from "react";
import { ImMenu } from "react-icons/im";
import { AiOutlineClose } from "react-icons/ai";
import { BiSearch } from "react-icons/bi";
import Content from "../../layouts/content";
import Navbar from "../Index/Navbar";
import "./index.scss";
import HomeLogo from "../../assets/img/logo";
import Transfer from "./items/transfer";
import Recurring from "./items/recurring";
import Delegate from "./items/delegate";
import PowerUp from "./items/power";
import Withdraw from "./items/withdraw";
import PowerDown from "./items/powerdown";
import TransferToSaving from "./items/transfertosaving";
import TransferFromSaving from "./items/transferfromsaving";
import CancelTransfer from "./items/canceltransfer";
import Convert from "./items/convert";
import Collateralized from "./items/collateralized";
import WitnessVote from "./items/witnessvote";
import WitnessProxy from "./items/witnessproxy";
import Claim from "./items/Claim";
import CreateAccount from "./items/createaccount";
import LimitOrderOne from "./items/Limit_Order_One";
import LimitOrderTwo from "./items/Limit_Order_Two";
import CancelLimitOrder from "./items/Cancel_Limit_Order";
import RedeemRewards from "./items/Redeem_Rewards";
import PostOrComment from "./items/Post_Or_Comment";
import PostOrCommentOptions from "./items/Post_Or_Comment_Options";
import CustomOperation from "./items/Custom_Operation";
import DeleteComment from "./items/Delete_Comment";
import AccountActive from "./items/Update_Account_Active";
import AccountPosting from "./items/Update_Account_Posting";
import RecoveryAccount from "./items/Recovery_Account";
import CreateProposal from "./items/Create_Proposal";
import RemoveProposal from "./items/Remove_Proposal";
import UpdateProposalVote from "./items/Update_Proposal_Votes";
import UpdateProposal from "./items/Update_Proposal";
import Vote from "./items/vote";
const listArr = [
  { key: 'transfer', value: <Transfer /> },
  { key: 'recurring transfers', value: <Recurring /> },
  { key: 'delegate hive power', value: <Delegate /> },
  { key: 'power up', value: <PowerUp /> },
  { key: 'set withdraw vesting route', value: <Withdraw /> },
  { key: 'powerdown', value: <PowerDown /> },
  { key: 'transfer to saving', value: <TransferToSaving /> },
  { key: 'transfer from saving', value: <TransferFromSaving /> },
  { key: 'cancel transfer from saving', value: <CancelTransfer /> },
  { key: 'convert', value: <Convert /> },
  { key: 'collateralized convert', value: <Collateralized /> },
  { key: 'witness vote', value: <WitnessVote /> },
  { key: 'witness proxy', value: <WitnessProxy /> },
  { key: 'claim account', value: <Claim /> },
  { key: 'create account', value: <CreateAccount /> },
  { key: 'vote', value: <Vote /> },
  { key: 'create limit order', value: <LimitOrderOne /> },
  { key: 'create limit order', value: <LimitOrderTwo /> },
  { key: 'cancel limit order', value: <CancelLimitOrder /> },
  { key: 'redeem rewards', value: <RedeemRewards /> },
  { key: 'post or comment', value: <PostOrComment /> },
  { key: 'post or comment options', value: <PostOrCommentOptions /> },
  { key: 'custom operation', value: <CustomOperation /> },
  { key: 'delete comment', value: <DeleteComment /> },
  { key: 'update account (active)', value: <AccountActive /> },
  { key: 'update account (posting)', value: <AccountPosting /> },
  { key: 'vhange recovery account', value: <RecoveryAccount /> },
  { key: 'vreate Proposal', value: <CreateProposal /> },
  { key: 'remove Proposal', value: <RemoveProposal /> },
  { key: 'update proposal votes', value: <UpdateProposalVote /> },
  { key: 'update proposal', value: <UpdateProposal /> },

];
const Index = () => {
  const [showNavbar, setshowNavbar] = useState(true);
  const [init_value, setInit_value] = useState("");
  const handleScreenChange = (mediaQuery: any) => {
    if (mediaQuery.matches) {
      setshowNavbar(true);
    } else {
      setshowNavbar(false);
    }
  };
  useEffect(() => {
    let window_width = window.matchMedia("(min-width:632px)");
    window_width.addListener(handleScreenChange);
    handleScreenChange(window_width);
    return () => {
      window_width.removeListener(handleScreenChange);
    };
  }, []);
  const handleChange = () => {
    setshowNavbar(!showNavbar);
  };
  const handleSearch = (e: any) => {
    setInit_value(e.target.value);
  };
  return (
    <Content>
      <div className="SignContainer">
        <div className="SignHeader">
          {/* <img src={HomeLogo} width="auto" height="auto" alt="logo" /> */}
          <div className="homelogo">
            <HomeLogo />
          </div>
          <div className="Title">
            <a href="#">Sign transaction</a>
          </div>
          <div className="SignMenubar" onClick={() => handleChange()}>
            {showNavbar ? <AiOutlineClose /> : <ImMenu />}
          </div>
        </div>
        <div>
          <div className="TransSearch">
            <input
              type="text"
              className="SignInput"
              value={init_value}
              onChange={(e: any) => handleSearch(e)}
              placeholder="Please type name of transaction that need to sign"
            />
            <BiSearch className="icon" />
          </div>
          <div className="ResultContent">
            {!init_value && listArr.map((v, k) => v.value)}
            {init_value && listArr.filter((v, k) => v.key.includes(init_value.toLowerCase())).map((v1, k1) => v1.value)}
          </div>
        </div>
      </div>
      <div className="Signfix">{showNavbar && <Navbar />}</div>
    </Content>
  );
};

export default Index;
