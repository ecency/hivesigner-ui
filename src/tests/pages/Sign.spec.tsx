import React from "react";
import Sign from "../../components/Sign";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
describe("Sign component Test", () => {
  Enzyme.configure({ adapter: new Adapter() });
  test("1 Snapshot testing", () => {
    const SignHeader = renderer.create(<div className="SignHeader" />).toJSON();
    expect(SignHeader).toMatchSnapshot();
  });
  test("2 Snapshot testing", () => {
    const TransSearch = renderer
      .create(<div className="TransSearch" />)
      .toJSON();
    expect(TransSearch).toMatchSnapshot();
  });
  test("3 Snapshot testing", () => {
    const ResultContent = renderer
      .create(<div className="ResultContent" />)
      .toJSON();
    expect(ResultContent).toMatchSnapshot();
  });
  test("4 Snapshot testing", () => {
    const Signfix = renderer.create(<div className="Signfix" />).toJSON();
    expect(Signfix).toMatchSnapshot();
  });

  test("each components test", () => {
    expect(shallow(<Sign />).find(".homelogo").length).toEqual(1);
    expect(shallow(<Sign />).find(".Title").length).toEqual(1);
    expect(shallow(<Sign />).find(".SignMenubar").length).toEqual(1);
  });
  test("function test", () => {
    const SignInput = jest.fn();
    const component = shallow(<Sign />);
    const event = {
      target: { value: "" },
    } as React.ChangeEvent<HTMLInputElement>;
    component.find(".SignInput").at(0).simulate("change", event);
    // expect(SignInput).toBeCalled();
  });
});
