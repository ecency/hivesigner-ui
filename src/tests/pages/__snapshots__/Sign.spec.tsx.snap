// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign component Test 1 Snapshot testing 1`] = `
<div
  className="SignHeader"
/>
`;

exports[`Sign component Test 2 Snapshot testing 1`] = `
<div
  className="TransSearch"
/>
`;

exports[`Sign component Test 3 Snapshot testing 1`] = `
<div
  className="ResultContent"
/>
`;

exports[`Sign component Test 4 Snapshot testing 1`] = `
<div
  className="Signfix"
/>
`;
